### PR TITLE
Avoid casting where possible

### PR DESCRIPTION
**Pull request by AtariDreams**
We should try to avoid changing the signedness of unsigned variables where possible.

### CODE REVIEW

File: instrument-functions.c [hunk 1]
```diff
@@ -158,14 +158,12 @@ static void print_debug(void *this_fn, void *call_site, action_type action)
 
 	if (instrument_global) {
 		symbol_info syminfo;
-		int found;
-
-		i = 0;
-		found = 0;
-		while (i < symcount && !found) {
+		int found = 0;
+		for (i = 0; i < symcount; i++) {
```
Comment (fxlb) [inline instrument-functions.c:162]: Any reason to change this file?
This change add bug.

Comment (AtariDreams) [reply instrument-functions.c:162]: It does not if we break immediately after?

**Comment by AtariDreams** [comment]
@fxlb I fixed the mistake.

**Comment by fxlb** [comment]
Your title is `Prefer unsigned variables where appropiate`.
Some changes are not related with this title.
What are the problems you want to solve?

**Comment by AtariDreams** [comment]
Avoid casting where possible.

**Comment by AtariDreams** [comment]
@fxlb I hope this works!

**Comment by infrastation** [comment]
The commit summary says "avoid casting", but the diff removes two type casts and adds three type casts. This needs to be a bit more focused.

File: instrument-functions.c [hunk 2]
```diff
@@ -158,16 +158,13 @@ static void print_debug(void *this_fn, void *call_site, action_type action)
...
```
Comment (fxlb) [inline instrument-functions.c:162]: Why this change?

Comment (AtariDreams) [reply instrument-functions.c:162]: It looks more intuitive. But if I must change it back I will.

**Comment by AtariDreams** [comment]
@infrastation Done!

File: instrument-functions.c (hunk 2 above)
Comment (AtariDreams) [reply instrument-functions.c]: @fxlb Changed back!

**Comment by AtariDreams** [comment]
@fxlb Done!

**Comment by infrastation** [comment]
As these changes are now, they still replace two type casts with three type casts and make a number of other unrelated changes, so I find it best to reject the proposed changes in order to be able to prioritize more important pending work. Thank you for your time anyway.
